staging/src/k8s.io/apiserver/pkg/storage/storagebackend/factory/etcd3.go

```diff
@@ -161,33 +179,43 @@ func newETCD3Check(c storagebackend.Config, timeout time.Duration, stopCh <-chan
 		defer utilruntime.HandleCrash()
 		<-stopCh
 
-		lock.Lock()
-		defer lock.Unlock()
+		lock.RLock()
+		defer lock.RUnlock()
```

Comment:
this is mutating variables, why is this not a write lock?

Reply:
IIUIC this lock is only for `clientErr`  ... but now that you mention it,  since we have the atomicError, we can use it now ,

Reply:
ahh, my bad, is locking the `client` too